dashboard/app/models/course_offering.rb

```diff
@@ -385,6 +386,8 @@ def self.properties_from_file(content)
   # seed_all
   def self.seed_record(file_path)
     properties = properties_from_file(File.read(file_path))
+    properties["self_paced_pl_course_offering_id"] = CourseOffering.find_by_key(properties["self_paced_pl_course_offering_key"])&.id
```

Comment:
one issue with this approach is going to be that you may have to seed twice in order for the `self_paced_pl_course_offering_id`s to be accurate, because the PL course offering may not have been seeded yet at the time a `self_paced_pl_course_offering_key` is being processed. I think this is probably going to be fine in levelbuilder/production, and most likely will only affect new engineers on our team who have just set up their local environment and are loading the curriculum catalog after only having seeded once.

my recommendation would be to detect this scenario and print out a warning message here when this happens, e.g. "self_paced_pl_course_offering_key #{key} not found. please seed again to fix."